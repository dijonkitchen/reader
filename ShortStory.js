import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
} from 'react-native';

const placeholderImage = require('./images/placeholder.png');

export default class ShortStory extends React.Component {
  render() {
    const { data } = this.props;
    let title = '';
    let description = '';
    if (data) {
      title = data.title;
      description = data.description.split('<')[0];
    }

    return (
      <View>
        <View style={styles.container}>
          <Image
            style={styles.thumbnails}
            source={placeholderImage}
          />
          <Text style={styles.title}>
            {title}
          </Text>
        </View>
        <Text style={styles.description}>
          {description}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  thumbnails: {
    flex: 1,
    height: 100,
    width: 100,
  },
  title: {
    flex: 2,
    fontWeight: 'bold',
    margin: 15,
  },
  description: {
    margin: 5,
  },
});
