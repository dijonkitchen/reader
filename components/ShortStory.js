import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableHighlight,
} from 'react-native';

const placeholderImage = require('../images/placeholder.png');

export default class ShortStory extends React.Component {
  render() {
    const { navigate } = this.props.navigation;
    const { data } = this.props;
    let title = '';
    let description = '';
    if (data) {
      title = data.title;
      description = data.description.split('<')[0];
    }

    return (
      <TouchableHighlight
        onPress={() => navigate('Story')}
      >
        <View>
          <View style={styles.container}>
            <Image
              style={styles.thumbnails}
              source={placeholderImage}
            />
            <Text style={styles.title}>
              {title}
            </Text>
          </View>
          <Text style={styles.description}>
            {description}
          </Text>
        </View>
      </TouchableHighlight>
    );
  }
}

ShortStory.navigationOptions = {
  title: 'ShortStory',
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  thumbnails: {
    flex: 1,
    height: 100,
    width: 100,
  },
  title: {
    flex: 2,
    fontWeight: 'bold',
    margin: 15,
  },
  description: {
    margin: 5,
  },
});
