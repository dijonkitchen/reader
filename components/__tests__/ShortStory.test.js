import React from 'react';
import {
  Image,
  View,
  Text,
  TouchableHighlight,
} from 'react-native';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import ShortStory from '../ShortStory';

describe('ShortStory component', () => {
  beforeEach(() => {
    wrapper = shallow(<ShortStory navigation={() => 'test'} />);
  });

  test('render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should be a certain component', () => {
    expect(wrapper.type()).toEqual(TouchableHighlight);
  });

  test('should be contain a Image component', () => {
    expect(wrapper.find(Image).length).toBe(1);
  });

  test('should be contain two Text components', () => {
    expect(wrapper.find(Text).length).toBe(2);
  });

  test('renders correctly', () => {
    const tree = renderer.create(
      <ShortStory navigation={() => 'test'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
