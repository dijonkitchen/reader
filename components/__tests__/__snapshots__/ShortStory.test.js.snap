// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortStory component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 100,
            "width": 100,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 2,
            "fontWeight": "bold",
            "margin": 15,
          }
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 5,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
